test/blackbox-tests/test-cases/melange/gh7020.t

```diff
@@ -53,4 +53,10 @@ Reproduce github #7020
   with Dune but published with a META template. Such libraries are not
   compatible with melange support
   -> required by alias mel
+  File "dune", line 4, characters 1-21:
+  4 |  (libraries dummyfoo)
+       ^^^^^^^^^^^^^^^^^^^^
+  Error: The library "dummyfoo" was added as a dependency of a melange.emit
+  stanza, but this library is not compatible with melange. To fix this, add
+  (modes melange) to the library stanza.
```

Comment:
```suggestion
  File `dune`, line 4, characters 1-21:
  4 |  (libraries `dummyfoo`)
       ^^^^^^^^^^^^^^^^^^^^
  Error: The library `dummyfoo` was added as a dependency of a `melange.emit`
  stanza, but this library is not compatible with Melange. To fix this, add
  (`modes melange`) to the library stanza.
```

Reply:
Thanks @christinerose. I updated the error style in https://github.com/ocaml/dune/pull/6412/commits/40c0fc4daef2a7eef5ca891bc113c64b0e4f2e45.

The ``"File `dune`", line...`` section is part of the generic error messages, and so a more broad change than the rest of the changes requested, I can open a new PR with that change so we can discuss it, because it will affect all user error messages.

I also wrapped the whole `(modes melange)` with backticks, because the parentheses are part of the Melange syntax. Please let me know if it looks ok, thanks!

Reply:
Looks great!